src/main/java/uk/gov/pay/api/model/PaymentWithLinks.java

```diff
@@ -3,10 +3,10 @@
 import java.net.URI;
 import java.util.List;
 
-public class PaymentWithLinks implements PaymentWithLinksJSON {
+public class PaymentWithLinks implements PaymentWithLinksJSON, PaymentJSON {
```

Comment:
I do agree the interfaces would be unnecessary given the current link structure we've agreed to. I've actually started working on getting rid of them in the next PR i'm creating (that fixes the PaymentEvents).
I was thinking in the lines of having a Payment object which has the API* annotations and Jackson annotations. 
But the links will be in their own classes. We can't have a generic '_links' attribute in the Payment class since Jackson and swagger needs the class structure to be in place in the class. In the case of having a '_self' and '_next_link', we have to have a object that is annotated with '_links' and we need to have '_self' and '_next_link' annotated inside that class. Else we don't get the same json structure we expect.
So essentially we'd have a hierarchy of Payment->PaymentWithSelfLink and Payment->PaymentWithSelfAndNextLink

** Check the comment below about naming